src/client/camera.cpp

```diff
@@ -462,33 +492,40 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 busytime, f32 tool_r
 		m_camera_position = my_cp;
 
 	/*
-	 * Apply server-sent FOV. If server doesn't enforce FOV,
-	 * check for zoom and set to zoom FOV.
-	 * Otherwise, default to m_cache_fov
+	 * Apply server-sent FOV, instantaneous or smooth transition.
+	 * If not, check for zoom and set to zoom FOV.
+	 * Otherwise, default to m_cache_fov.
 	 */
-
-	f32 fov_degrees;
-	PlayerFovSpec fov_spec = player->getFov();
-	if (fov_spec.fov > 0.0f) {
-		// If server-sent FOV is a multiplier, multiply
-		// it with m_cache_fov instead of overriding
-		if (fov_spec.is_multiplier)
-			fov_degrees = m_cache_fov * fov_spec.fov;
-		else
-			fov_degrees = fov_spec.fov;
+	if (m_fov_transition_active) {
+		// Smooth FOV transition
+		// Dynamically calculate FOV delta based on frametimes
+		f32 delta = (frametime / m_transition_time) * m_fov_diff;
+		m_curr_fov_degrees += delta;
+
+		// Mark transition as complete if target FOV has been reached
+		if ((m_fov_diff > 0 && m_curr_fov_degrees >= m_server_sent_fov) ||
+				(m_fov_diff < 0 && m_curr_fov_degrees <= m_server_sent_fov)) {
+			m_fov_transition_active = false;
+
+			// Update m_old_fov_degrees to prepare for next FOV transition
+			m_old_fov_degrees = m_curr_fov_degrees = m_server_sent_fov;
+		}
+	} else if (m_server_sent_fov > 0.0f) {
+		// Instantaneous FOV change
+		m_old_fov_degrees = m_curr_fov_degrees = m_server_sent_fov;
 	} else if (player->getPlayerControl().zoom && player->getZoomFOV() > 0.001f) {
 		// Player requests zoom, apply zoom FOV
-		fov_degrees = player->getZoomFOV();
+		m_curr_fov_degrees = player->getZoomFOV();
 	} else {
 		// Set to client's selected FOV
-		fov_degrees = m_cache_fov;
+		m_old_fov_degrees = m_curr_fov_degrees = m_cache_fov;
```

Comment:
Why is `m_old_fov_degrees` set everywhere? it should only be set by `notifyFovChange()` if a transition actually happens

Reply:
This isn't actually required now, AFAIK. I'll remove this and test various scenarios again.

Reply:
Simply removing these seems to break a lot of stuff. I believe this is because a new transition doesn't have much information about the previous transition. Since `m_old_fov_degrees` corresponds to the target FOV of the previous transition, setting it in `notifyFovChange` won't work because there's no way to access the final target FOV of the previous transition from here.

I'll try to see if I can work around this, because I agree that this shouldn't be set everywhere.

Reply:
Got it to work again by tweaking `notifyFovChange`. I can now see that the old code was pretty messy, lol.